bundlewrap/items/routeros.py

```diff
@@ -26,30 +29,68 @@ def __repr__(self):
     def cdict(self):
         if self.attributes['delete']:
             return None
+
+        if self.attributes['purge']:
+            # purge operates on lists of items only which we can't operate on otherwise
+            return {
+                'subitems_to_purge': set()
+            }
+
         cdict = self.attributes.copy()
         if '_comment' in cdict:  # work around 'comment' being a builtin attribute
             cdict['comment'] = cdict['_comment']
             del cdict['_comment']
+
         del cdict['delete']
+        del cdict['purge']
         return cdict
 
+    def get_basename(self, name):
+        return name.split("?", 1)[0]
+
+    @property
+    def basename(self):
+        return self.get_basename(self.name)
```

Comment:
Is there any reason not to merge those two methods? `get_basename()` is not used anywhere else? (same for `get_identifier()` below.

Reply:
`get_identifier` *is* used further down in `_get_subitems_to_purge` and I split up ´basename` for symmetry, too.